src/network/networkctl.c

```diff
@@ -958,12 +959,18 @@ static int dump_addresses(
                         return r;
 
                 if (local[i].family == AF_INET && in4_addr_equal(&local[i].address.in, &dhcp4_address)) {
-                        _cleanup_free_ char *p = NULL;
+                        struct in_addr server_address;
+                        char *p, s[INET_ADDRSTRLEN];
 
-                        p = pretty;
-                        pretty = strjoin(pretty , " (DHCP4)");
-                        if (!pretty)
+                        r = sd_dhcp_lease_get_server_identifier(lease, &server_address);
+                        if (r >= 0)
+                                p = strjoin(pretty, " (DHCP4 via ", inet_ntop(AF_INET, &server_address, s, sizeof(s)), ")");
+                        else
+                                p = strjoin(pretty, " (DHCP4)");
+                        if (!p)
```

Comment:
I think this situation will never happen. Without DHCP server client wont get a IP. But I think OK.

Reply:
But, reading lease file may fail. Just for safety.

Reply:
Unless someone deletes the lease between which is highly unlikely. But ok.